<h1> TEST</h1>

In [223]:
print('Hello World')

Hello World


<h1> DOWNLOAD Libraries</h1>

In [224]:
pip install numpy

Note: you may need to restart the kernel to use updated packages.


In [225]:
pip install pandas

Note: you may need to restart the kernel to use updated packages.


In [226]:
pip install matplotlib

Note: you may need to restart the kernel to use updated packages.


In [227]:
pip install seaborn

Note: you may need to restart the kernel to use updated packages.


In [228]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

<h1>IMPORTANT DATASET TO DATA FRAME</h1>

In [229]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [230]:
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
3        4     Adelie  Torgersen             NaN            NaN   
4        5     Adelie  Torgersen            36.7           19.3   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[344 rows x 9 columns]

<h1> Checking your DATA FRAME for null values, and data types</h1>

In [231]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   rowid              344 non-null    int64  
 1   species            344 non-null    object 
 2   island             344 non-null    object 
 3   bill_length_mm     342 non-null    float64
 4   bill_depth_mm      342 non-null    float64
 5   flipper_length_mm  342 non-null    float64
 6   body_mass_g        342 non-null    float64
 7   sex                333 non-null    object 
 8   year               344 non-null    int64  
dtypes: float64(4), int64(2), object(3)
memory usage: 24.3+ KB


In [232]:
penguins.isnull()

rowid  species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    False    False   False           False          False              False   
1    False    False   False           False          False              False   
2    False    False   False           False          False              False   
3    False    False   False            True           True               True   
4    False    False   False           False          False              False   
..     ...      ...     ...             ...            ...                ...   
339  False    False   False           False          False              False   
340  False    False   False           False          False              False   
341  False    False   False           False          False              False   
342  False    False   False           False          False              False   
343  False    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 9 columns]

In [233]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h1>DESCRIPTIVE ANALYTICS</h1>

In [234]:
penguins.describe()

rowid  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count  344.000000      342.000000     342.000000         342.000000   
mean   172.500000       43.921930      17.151170         200.915205   
std     99.448479        5.459584       1.974793          14.061714   
min      1.000000       32.100000      13.100000         172.000000   
25%     86.750000       39.225000      15.600000         190.000000   
50%    172.500000       44.450000      17.300000         197.000000   
75%    258.250000       48.500000      18.700000         213.000000   
max    344.000000       59.600000      21.500000         231.000000   

       body_mass_g         year  
count   342.000000   344.000000  
mean   4201.754386  2008.029070  
std     801.954536     0.818356  
min    2700.000000  2007.000000  
25%    3550.000000  2007.000000  
50%    4050.000000  2008.000000  
75%    4750.000000  2009.000000  
max    6300.000000  2009.000000

In [235]:
#to flip the describe table

In [236]:
penguins.describe().T

count         mean         std     min       25%      50%  \
rowid              344.0   172.500000   99.448479     1.0    86.750   172.50   
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                       75%     max  
rowid               258.25   344.0  
bill_length_mm       48.50    59.6  
bill_depth_mm        18.70    21.5  
flipper_length_mm   213.00   231.0  
body_mass_g        4750.00  6300.0  
year               2009.00  2009.0

<h1>FILTERING DATA BASED FROM COLUMNS</h1>

In [237]:
penguins['body_mass_g']

0      3750.0
1      3800.0
2      3250.0
3         NaN
4      3450.0
        ...  
339    4000.0
340    3400.0
341    3775.0
342    4100.0
343    3775.0
Name: body_mass_g, Length: 344, dtype: float64

<h1>HOW TO DEFINE/ CREATE YOUR OWN COLUMN</h1>

In [238]:
penguins['weight']= np.round(penguins['body_mass_g'] / 1000,2)

In [239]:
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
3        4     Adelie  Torgersen             NaN            NaN   
4        5     Adelie  Torgersen            36.7           19.3   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  weight  
0                181.0       3750.0    male  2007    3.75  
1                186.0       3800.0  female  2007    3.80  
2                195.0       3250.0  female  2007    3.25  
3                  NaN          NaN     NaN  2007     NaN  
4                193.0       3450.0  female  2007    3.45  
..                 ...          ...     ...   ...     ...  
339              207.0       4000.0    male  2009    4.00  
340              202.0       3400.0  female  2009    3.40  
341              193.0       3775.0    male  2009    3.78  
342              210.0       4100.0    male  2009    4.10  
343              198.0       3775.0  female  2009    3.78  

[344 rows x 10 columns]

In [240]:
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
3        4     Adelie  Torgersen             NaN            NaN   
4        5     Adelie  Torgersen            36.7           19.3   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  weight  
0                181.0       3750.0    male  2007    3.75  
1                186.0       3800.0  female  2007    3.80  
2                195.0       3250.0  female  2007    3.25  
3                  NaN          NaN     NaN  2007     NaN  
4                193.0       3450.0  female  2007    3.45  
..                 ...          ...     ...   ...     ...  
339              207.0       4000.0    male  2009    4.00  
340              202.0       3400.0  female  2009    3.40  
341              193.0       3775.0    male  2009    3.78  
342              210.0       4100.0    male  2009    4.10  
343              198.0       3775.0  female  2009    3.78  

[344 rows x 10 columns]

<h1>WHAT TO DO WITH NULL VALUES</h1>

delete the row

fill the empty cells with a value

In [241]:
penguins['body_mass_g'].mean()

np.float64(4201.754385964912)

In [242]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

In [243]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [244]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [245]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           0
sex                  11
year                  0
weight                2
dtype: int64

In [246]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [247]:
penguins.isnull().sum()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        2
flipper_length_mm    2
body_mass_g          0
sex                  0
year                 0
weight               2
dtype: int64

In [248]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [249]:
penguins.isnull().sum()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    2
body_mass_g          0
sex                  0
year                 0
weight               2
dtype: int64

In [250]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [251]:
penguins.isnull().sum()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
weight               2
dtype: int64

In [252]:
penguins['weight'] = penguins['weight'].fillna(penguins['weight'].mean())

In [253]:
penguins.isnull().sum()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
weight               0
dtype: int64

In [254]:
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen        39.10000       18.70000   
1        2     Adelie  Torgersen        39.50000       17.40000   
2        3     Adelie  Torgersen        40.30000       18.00000   
3        4     Adelie  Torgersen        43.92193       17.15117   
4        5     Adelie  Torgersen        36.70000       19.30000   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream        55.80000       19.80000   
340    341  Chinstrap      Dream        43.50000       18.10000   
341    342  Chinstrap      Dream        49.60000       18.20000   
342    343  Chinstrap      Dream        50.80000       19.00000   
343    344  Chinstrap      Dream        50.20000       18.70000   

     flipper_length_mm  body_mass_g     sex  year    weight  
0           181.000000  3750.000000    male  2007  3.750000  
1           186.000000  3800.000000  female  2007  3.800000  
2           195.000000  3250.000000  female  2007  3.250000  
3           200.915205  4201.754386    male  2007  4.201784  
4           193.000000  3450.000000  female  2007  3.450000  
..                 ...          ...     ...   ...       ...  
339         207.000000  4000.000000    male  2009  4.000000  
340         202.000000  3400.000000  female  2009  3.400000  
341         193.000000  3775.000000    male  2009  3.780000  
342         210.000000  4100.000000    male  2009  4.100000  
343         198.000000  3775.000000  female  2009  3.780000  

[344 rows x 10 columns]

In [255]:
penguins.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   rowid              344 non-null    int64  
 1   species            344 non-null    object 
 2   island             344 non-null    object 
 3   bill_length_mm     344 non-null    float64
 4   bill_depth_mm      344 non-null    float64
 5   flipper_length_mm  344 non-null    float64
 6   body_mass_g        344 non-null    float64
 7   sex                344 non-null    object 
 8   year               344 non-null    int64  
 9   weight             344 non-null    float64
dtypes: float64(5), int64(2), object(3)
memory usage: 27.0+ KB


In [256]:
penguins['year'] = penguins['year'].astype('category')

In [257]:
penguins['species'] = penguins['species'].astype('category')

In [258]:
penguins['rowid'] = penguins['rowid'].astype('category')

In [259]:
penguins['island'] = penguins['island'].astype('category')

In [260]:
penguins['sex'] = penguins['sex'].astype('category')

In [261]:
penguins.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              344 non-null    category
 1   species            344 non-null    category
 2   island             344 non-null    category
 3   bill_length_mm     344 non-null    float64 
 4   bill_depth_mm      344 non-null    float64 
 5   flipper_length_mm  344 non-null    float64 
 6   body_mass_g        344 non-null    float64 
 7   sex                344 non-null    category
 8   year               344 non-null    category
 9   weight             344 non-null    float64 
dtypes: category(5), float64(5)
memory usage: 26.9 KB


In [262]:
penguins['species'] == "Chinstrap"

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

In [263]:
penguins[penguins['species'] == "Chinstrap"].count()

rowid                68
species              68
island               68
bill_length_mm       68
bill_depth_mm        68
flipper_length_mm    68
body_mass_g          68
sex                  68
year                 68
weight               68
dtype: int64

In [264]:
penguins[penguins['species'] == "Adelie"].count()

rowid                152
species              152
island               152
bill_length_mm       152
bill_depth_mm        152
flipper_length_mm    152
body_mass_g          152
sex                  152
year                 152
weight               152
dtype: int64

In [265]:
penguins[penguins['species'] == "Gentoo"].count()

rowid                124
species              124
island               124
bill_length_mm       124
bill_depth_mm        124
flipper_length_mm    124
body_mass_g          124
sex                  124
year                 124
weight               124
dtype: int64

<h1>HOW MANY GENTOO SPECIES GAVE WEIGHT MORE THAN 3KG</h1>

In [266]:
penguins[penguins['species'] == "Gentoo"]

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152   153  Gentoo  Biscoe        46.10000       13.20000         211.000000   
153   154  Gentoo  Biscoe        50.00000       16.30000         230.000000   
154   155  Gentoo  Biscoe        48.70000       14.10000         210.000000   
155   156  Gentoo  Biscoe        50.00000       15.20000         218.000000   
156   157  Gentoo  Biscoe        47.60000       14.50000         215.000000   
..    ...     ...     ...             ...            ...                ...   
271   272  Gentoo  Biscoe        43.92193       17.15117         200.915205   
272   273  Gentoo  Biscoe        46.80000       14.30000         215.000000   
273   274  Gentoo  Biscoe        50.40000       15.70000         222.000000   
274   275  Gentoo  Biscoe        45.20000       14.80000         212.000000   
275   276  Gentoo  Biscoe        49.90000       16.10000         213.000000   

     body_mass_g     sex  year    weight  
152  4500.000000  female  2007  4.500000  
153  5700.000000    male  2007  5.700000  
154  4450.000000  female  2007  4.450000  
155  5700.000000    male  2007  5.700000  
156  5400.000000    male  2007  5.400000  
..           ...     ...   ...       ...  
271  4201.754386    male  2009  4.201784  
272  4850.000000  female  2009  4.850000  
273  5750.000000    male  2009  5.750000  
274  5200.000000  female  2009  5.200000  
275  5400.000000    male  2009  5.400000  

[124 rows x 10 columns]

In [267]:
penguins[penguins['weight'] > 3].count()

rowid                333
species              333
island               333
bill_length_mm       333
bill_depth_mm        333
flipper_length_mm    333
body_mass_g          333
sex                  333
year                 333
weight               333
dtype: int64

In [268]:
penguins[(penguins['species'] == "Gentoo") & (penguins['weight'] > 3)]

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152   153  Gentoo  Biscoe        46.10000       13.20000         211.000000   
153   154  Gentoo  Biscoe        50.00000       16.30000         230.000000   
154   155  Gentoo  Biscoe        48.70000       14.10000         210.000000   
155   156  Gentoo  Biscoe        50.00000       15.20000         218.000000   
156   157  Gentoo  Biscoe        47.60000       14.50000         215.000000   
..    ...     ...     ...             ...            ...                ...   
271   272  Gentoo  Biscoe        43.92193       17.15117         200.915205   
272   273  Gentoo  Biscoe        46.80000       14.30000         215.000000   
273   274  Gentoo  Biscoe        50.40000       15.70000         222.000000   
274   275  Gentoo  Biscoe        45.20000       14.80000         212.000000   
275   276  Gentoo  Biscoe        49.90000       16.10000         213.000000   

     body_mass_g     sex  year    weight  
152  4500.000000  female  2007  4.500000  
153  5700.000000    male  2007  5.700000  
154  4450.000000  female  2007  4.450000  
155  5700.000000    male  2007  5.700000  
156  5400.000000    male  2007  5.400000  
..           ...     ...   ...       ...  
271  4201.754386    male  2009  4.201784  
272  4850.000000  female  2009  4.850000  
273  5750.000000    male  2009  5.750000  
274  5200.000000  female  2009  5.200000  
275  5400.000000    male  2009  5.400000  

[124 rows x 10 columns]

<h1>HOW MANY GENTOO SPECIES GAVE WEIGHT MORE THAN 3KG</h1>

In [269]:
penguins[(penguins['species'] == "Gentoo") & (penguins['weight'] < 3)].count()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
weight               0
dtype: int64

<h1>Does Gentoo exclusively live on just one ISLAND</h1>

In [270]:
penguins['species'] == "Gentoo"

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Name: species, Length: 344, dtype: bool

In [271]:
penguins[penguins['species'] == "Gentoo"]

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152   153  Gentoo  Biscoe        46.10000       13.20000         211.000000   
153   154  Gentoo  Biscoe        50.00000       16.30000         230.000000   
154   155  Gentoo  Biscoe        48.70000       14.10000         210.000000   
155   156  Gentoo  Biscoe        50.00000       15.20000         218.000000   
156   157  Gentoo  Biscoe        47.60000       14.50000         215.000000   
..    ...     ...     ...             ...            ...                ...   
271   272  Gentoo  Biscoe        43.92193       17.15117         200.915205   
272   273  Gentoo  Biscoe        46.80000       14.30000         215.000000   
273   274  Gentoo  Biscoe        50.40000       15.70000         222.000000   
274   275  Gentoo  Biscoe        45.20000       14.80000         212.000000   
275   276  Gentoo  Biscoe        49.90000       16.10000         213.000000   

     body_mass_g     sex  year    weight  
152  4500.000000  female  2007  4.500000  
153  5700.000000    male  2007  5.700000  
154  4450.000000  female  2007  4.450000  
155  5700.000000    male  2007  5.700000  
156  5400.000000    male  2007  5.400000  
..           ...     ...   ...       ...  
271  4201.754386    male  2009  4.201784  
272  4850.000000  female  2009  4.850000  
273  5750.000000    male  2009  5.750000  
274  5200.000000  female  2009  5.200000  
275  5400.000000    male  2009  5.400000  

[124 rows x 10 columns]

<h1>
    GENTOO IN BISCOE ISLAND
</h1>

In [272]:
gentoo_biscoe = penguins[(penguins['island'] == "Biscoe") & (penguins['species'] == "Gentoo")] 

In [273]:
gentoo_biscoe

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152   153  Gentoo  Biscoe        46.10000       13.20000         211.000000   
153   154  Gentoo  Biscoe        50.00000       16.30000         230.000000   
154   155  Gentoo  Biscoe        48.70000       14.10000         210.000000   
155   156  Gentoo  Biscoe        50.00000       15.20000         218.000000   
156   157  Gentoo  Biscoe        47.60000       14.50000         215.000000   
..    ...     ...     ...             ...            ...                ...   
271   272  Gentoo  Biscoe        43.92193       17.15117         200.915205   
272   273  Gentoo  Biscoe        46.80000       14.30000         215.000000   
273   274  Gentoo  Biscoe        50.40000       15.70000         222.000000   
274   275  Gentoo  Biscoe        45.20000       14.80000         212.000000   
275   276  Gentoo  Biscoe        49.90000       16.10000         213.000000   

     body_mass_g     sex  year    weight  
152  4500.000000  female  2007  4.500000  
153  5700.000000    male  2007  5.700000  
154  4450.000000  female  2007  4.450000  
155  5700.000000    male  2007  5.700000  
156  5400.000000    male  2007  5.400000  
..           ...     ...   ...       ...  
271  4201.754386    male  2009  4.201784  
272  4850.000000  female  2009  4.850000  
273  5750.000000    male  2009  5.750000  
274  5200.000000  female  2009  5.200000  
275  5400.000000    male  2009  5.400000  

[124 rows x 10 columns]

In [274]:
penguins[(penguins['species'] == "Gentoo") & (penguins['island'] == "Biscoe")].count()

rowid                124
species              124
island               124
bill_length_mm       124
bill_depth_mm        124
flipper_length_mm    124
body_mass_g          124
sex                  124
year                 124
weight               124
dtype: int64

<h1>
    GENTOO IN DREAM ISLAND
</h1>

In [275]:
gentoo_dream = penguins[(penguins['island'] == "Dream") & (penguins['species'] == "Gentoo")] 

In [276]:
gentoo_dream

Empty DataFrame
Columns: [rowid, species, island, bill_length_mm, bill_depth_mm, flipper_length_mm, body_mass_g, sex, year, weight]
Index: []

In [277]:
penguins[(penguins['species'] == "Gentoo") & (penguins['island'] == "Dream")].count()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
weight               0
dtype: int64

<h1>
    GENTOO IN Torgersen ISLAND
</h1>

In [278]:
gentoo_torgersen = penguins[(penguins['island'] == "Torgersen") & (penguins['species'] == "Gentoo")] 

In [279]:
gentoo_torgersen

Empty DataFrame
Columns: [rowid, species, island, bill_length_mm, bill_depth_mm, flipper_length_mm, body_mass_g, sex, year, weight]
Index: []

In [280]:
penguins[(penguins['species'] == "Gentoo") & (penguins['island'] == "Dream")].count()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
weight               0
dtype: int64

<h1>INSIGHT NO.1</h1>

<H3> GENTOO IS EXCLUSIVLY LIVE IN BISCOE</H3>


<h1>
    ADELIE IN BISCOE ISLAND
</h1>

In [281]:
adelie_Biscoe = penguins[(penguins['island'] == "Biscoe") & (penguins['species'] == "Adelie")] 

In [282]:
adelie_Biscoe

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
20     21  Adelie  Biscoe            37.8           18.3              174.0   
21     22  Adelie  Biscoe            37.7           18.7              180.0   
22     23  Adelie  Biscoe            35.9           19.2              189.0   
23     24  Adelie  Biscoe            38.2           18.1              185.0   
24     25  Adelie  Biscoe            38.8           17.2              180.0   
25     26  Adelie  Biscoe            35.3           18.9              187.0   
26     27  Adelie  Biscoe            40.6           18.6              183.0   
27     28  Adelie  Biscoe            40.5           17.9              187.0   
28     29  Adelie  Biscoe            37.9           18.6              172.0   
29     30  Adelie  Biscoe            40.5           18.9              180.0   
50     51  Adelie  Biscoe            39.6           17.7              186.0   
51     52  Adelie  Biscoe            40.1           18.9              188.0   
52     53  Adelie  Biscoe            35.0           17.9              190.0   
53     54  Adelie  Biscoe            42.0           19.5              200.0   
54     55  Adelie  Biscoe            34.5           18.1              187.0   
55     56  Adelie  Biscoe            41.4           18.6              191.0   
56     57  Adelie  Biscoe            39.0           17.5              186.0   
57     58  Adelie  Biscoe            40.6           18.8              193.0   
58     59  Adelie  Biscoe            36.5           16.6              181.0   
59     60  Adelie  Biscoe            37.6           19.1              194.0   
60     61  Adelie  Biscoe            35.7           16.9              185.0   
61     62  Adelie  Biscoe            41.3           21.1              195.0   
62     63  Adelie  Biscoe            37.6           17.0              185.0   
63     64  Adelie  Biscoe            41.1           18.2              192.0   
64     65  Adelie  Biscoe            36.4           17.1              184.0   
65     66  Adelie  Biscoe            41.6           18.0              192.0   
66     67  Adelie  Biscoe            35.5           16.2              195.0   
67     68  Adelie  Biscoe            41.1           19.1              188.0   
100   101  Adelie  Biscoe            35.0           17.9              192.0   
101   102  Adelie  Biscoe            41.0           20.0              203.0   
102   103  Adelie  Biscoe            37.7           16.0              183.0   
103   104  Adelie  Biscoe            37.8           20.0              190.0   
104   105  Adelie  Biscoe            37.9           18.6              193.0   
105   106  Adelie  Biscoe            39.7           18.9              184.0   
106   107  Adelie  Biscoe            38.6           17.2              199.0   
107   108  Adelie  Biscoe            38.2           20.0              190.0   
108   109  Adelie  Biscoe            38.1           17.0              181.0   
109   110  Adelie  Biscoe            43.2           19.0              197.0   
110   111  Adelie  Biscoe            38.1           16.5              198.0   
111   112  Adelie  Biscoe            45.6           20.3              191.0   
112   113  Adelie  Biscoe            39.7           17.7              193.0   
113   114  Adelie  Biscoe            42.2           19.5              197.0   
114   115  Adelie  Biscoe            39.6           20.7              191.0   
115   116  Adelie  Biscoe            42.7           18.3              196.0   

     body_mass_g     sex  year  weight  
20        3400.0  female  2007    3.40  
21        3600.0    male  2007    3.60  
22        3800.0  female  2007    3.80  
23        3950.0    male  2007    3.95  
24        3800.0    male  2007    3.80  
25        3800.0  female  2007    3.80  
26        3550.0    male  2007    3.55  
27        3200.0  female  2007    3.20  
28        3150.0  female  2007    3.15  
29        3950.0    male  2007    3.95

In [283]:
penguins[(penguins['species'] == "Adelie") & (penguins['island'] == "Biscoe")].count()

rowid                44
species              44
island               44
bill_length_mm       44
bill_depth_mm        44
flipper_length_mm    44
body_mass_g          44
sex                  44
year                 44
weight               44
dtype: int64

<h1>
    ADELIE IN DREAM ISLAND
</h1>

In [284]:
adelie_dream = penguins[(penguins['island'] == "Dream") & (penguins['species'] == "Adelie")] 

In [285]:
adelie_dream

rowid species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30     31  Adelie  Dream            39.5           16.7              178.0   
31     32  Adelie  Dream            37.2           18.1              178.0   
32     33  Adelie  Dream            39.5           17.8              188.0   
33     34  Adelie  Dream            40.9           18.9              184.0   
34     35  Adelie  Dream            36.4           17.0              195.0   
35     36  Adelie  Dream            39.2           21.1              196.0   
36     37  Adelie  Dream            38.8           20.0              190.0   
37     38  Adelie  Dream            42.2           18.5              180.0   
38     39  Adelie  Dream            37.6           19.3              181.0   
39     40  Adelie  Dream            39.8           19.1              184.0   
40     41  Adelie  Dream            36.5           18.0              182.0   
41     42  Adelie  Dream            40.8           18.4              195.0   
42     43  Adelie  Dream            36.0           18.5              186.0   
43     44  Adelie  Dream            44.1           19.7              196.0   
44     45  Adelie  Dream            37.0           16.9              185.0   
45     46  Adelie  Dream            39.6           18.8              190.0   
46     47  Adelie  Dream            41.1           19.0              182.0   
47     48  Adelie  Dream            37.5           18.9              179.0   
48     49  Adelie  Dream            36.0           17.9              190.0   
49     50  Adelie  Dream            42.3           21.2              191.0   
84     85  Adelie  Dream            37.3           17.8              191.0   
85     86  Adelie  Dream            41.3           20.3              194.0   
86     87  Adelie  Dream            36.3           19.5              190.0   
87     88  Adelie  Dream            36.9           18.6              189.0   
88     89  Adelie  Dream            38.3           19.2              189.0   
89     90  Adelie  Dream            38.9           18.8              190.0   
90     91  Adelie  Dream            35.7           18.0              202.0   
91     92  Adelie  Dream            41.1           18.1              205.0   
92     93  Adelie  Dream            34.0           17.1              185.0   
93     94  Adelie  Dream            39.6           18.1              186.0   
94     95  Adelie  Dream            36.2           17.3              187.0   
95     96  Adelie  Dream            40.8           18.9              208.0   
96     97  Adelie  Dream            38.1           18.6              190.0   
97     98  Adelie  Dream            40.3           18.5              196.0   
98     99  Adelie  Dream            33.1           16.1              178.0   
99    100  Adelie  Dream            43.2           18.5              192.0   
132   133  Adelie  Dream            36.8           18.5              193.0   
133   134  Adelie  Dream            37.5           18.5              199.0   
134   135  Adelie  Dream            38.1           17.6              187.0   
135   136  Adelie  Dream            41.1           17.5              190.0   
136   137  Adelie  Dream            35.6           17.5              191.0   
137   138  Adelie  Dream            40.2           20.1              200.0   
138   139  Adelie  Dream            37.0           16.5              185.0   
139   140  Adelie  Dream            39.7           17.9              193.0   
140   141  Adelie  Dream            40.2           17.1              193.0   
141   142  Adelie  Dream            40.6           17.2              187.0   
142   143  Adelie  Dream            32.1           15.5              188.0   
143   144  Adelie  Dream            40.7           17.0              190.0   
144   145  Adelie  Dream            37.3           16.8              192.0   
145   146  Adelie  Dream            39.0           18.7              185.0   
146   147  Adelie  Dream  

In [286]:
penguins[(penguins['species'] == "Adelie") & (penguins['island'] == "Dream")].count()

rowid                56
species              56
island               56
bill_length_mm       56
bill_depth_mm        56
flipper_length_mm    56
body_mass_g          56
sex                  56
year                 56
weight               56
dtype: int64

<h1>
    ADELIE IN TORGERSEN ISLAND
</h1>

In [287]:
adelie_torgersen = penguins[(penguins['island'] == "Torgersen") & (penguins['species'] == "Adelie")] 

In [288]:
adelie_torgersen

rowid species     island  bill_length_mm  bill_depth_mm  \
0       1  Adelie  Torgersen        39.10000       18.70000   
1       2  Adelie  Torgersen        39.50000       17.40000   
2       3  Adelie  Torgersen        40.30000       18.00000   
3       4  Adelie  Torgersen        43.92193       17.15117   
4       5  Adelie  Torgersen        36.70000       19.30000   
5       6  Adelie  Torgersen        39.30000       20.60000   
6       7  Adelie  Torgersen        38.90000       17.80000   
7       8  Adelie  Torgersen        39.20000       19.60000   
8       9  Adelie  Torgersen        34.10000       18.10000   
9      10  Adelie  Torgersen        42.00000       20.20000   
10     11  Adelie  Torgersen        37.80000       17.10000   
11     12  Adelie  Torgersen        37.80000       17.30000   
12     13  Adelie  Torgersen        41.10000       17.60000   
13     14  Adelie  Torgersen        38.60000       21.20000   
14     15  Adelie  Torgersen        34.60000       21.10000   
15     16  Adelie  Torgersen        36.60000       17.80000   
16     17  Adelie  Torgersen        38.70000       19.00000   
17     18  Adelie  Torgersen        42.50000       20.70000   
18     19  Adelie  Torgersen        34.40000       18.40000   
19     20  Adelie  Torgersen        46.00000       21.50000   
68     69  Adelie  Torgersen        35.90000       16.60000   
69     70  Adelie  Torgersen        41.80000       19.40000   
70     71  Adelie  Torgersen        33.50000       19.00000   
71     72  Adelie  Torgersen        39.70000       18.40000   
72     73  Adelie  Torgersen        39.60000       17.20000   
73     74  Adelie  Torgersen        45.80000       18.90000   
74     75  Adelie  Torgersen        35.50000       17.50000   
75     76  Adelie  Torgersen        42.80000       18.50000   
76     77  Adelie  Torgersen        40.90000       16.80000   
77     78  Adelie  Torgersen        37.20000       19.40000   
78     79  Adelie  Torgersen        36.20000       16.10000   
79     80  Adelie  Torgersen        42.10000       19.10000   
80     81  Adelie  Torgersen        34.60000       17.20000   
81     82  Adelie  Torgersen        42.90000       17.60000   
82     83  Adelie  Torgersen        36.70000       18.80000   
83     84  Adelie  Torgersen        35.10000       19.40000   
116   117  Adelie  Torgersen        38.60000       17.00000   
117   118  Adelie  Torgersen        37.30000       20.50000   
118   119  Adelie  Torgersen        35.70000       17.00000   
119   120  Adelie  Torgersen        41.10000       18.60000   
120   121  Adelie  Torgersen        36.20000       17.20000   
121   122  Adelie  Torgersen        37.70000       19.80000   
122   123  Adelie  Torgersen        40.20000       17.00000   
123   124  Adelie  Torgersen        41.40000       18.50000   
124   125  Adelie  Torgersen        35.20000       15.90000   
125   126  Adelie  Torgersen        40.60000       19.00000   
126   127  Adelie  Torgersen        38.80000       17.60000   
127   128  Adelie  Torgersen        41.50000       18.30000   
128   129  Adelie  Torgersen        39.00000       17.10000   
129   130  Adelie  Torgersen        44.10000       18.00000   
130   131  Adelie  Torgersen        38.50000       17.90000   
131   132  Adelie  Torgersen        43.10000       19.20000   

     flipper_length_mm  body_mass_g     sex  year    weight  
0           181.000000  3750.000000    male  2007  3.750000  
1           186.000000  3800.000000  female  2007  3.800000  
2           195.000000  3250.000000  female  2007  3.250000  
3           200.915205  4201.754386    male  2007  4.201784  
4           193.000000  3450.000000  female  2007  3.450000  
5           190.000000  3650.000000    male  2007  3.650000  
6           181.000000  3625.000000  female  2007  3.620000  
7           195.000000  4675.000000    male  2007  4.680000  
8           193.000000  3475.000000    male  2007  3.480000  
9           190.000000  4250.000000    male 

In [289]:
penguins[(penguins['species'] == "Adelie") & (penguins['island'] == "Torgersen")].count()

rowid                52
species              52
island               52
bill_length_mm       52
bill_depth_mm        52
flipper_length_mm    52
body_mass_g          52
sex                  52
year                 52
weight               52
dtype: int64

<h1>INSIGHT NO.2</h1>

<H3> ADELIE LIVES IN 3 ISLAND</H3>


In [290]:
chinstrap_biscoe = penguins[(penguins['island'] == "Biscoe") & (penguins['species'] == "Chinstrap")] 

In [291]:
chinstrap_biscoe

Empty DataFrame
Columns: [rowid, species, island, bill_length_mm, bill_depth_mm, flipper_length_mm, body_mass_g, sex, year, weight]
Index: []

In [292]:
penguins[(penguins['species'] == "Chinstrap") & (penguins['island'] == "Biscoe")].count()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
weight               0
dtype: int64

In [293]:
chinstrap_dream = penguins[(penguins['island'] == "Dream") & (penguins['species'] == "Chinstrap")] 

In [294]:
chinstrap_dream

rowid    species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276   277  Chinstrap  Dream            46.5           17.9              192.0   
277   278  Chinstrap  Dream            50.0           19.5              196.0   
278   279  Chinstrap  Dream            51.3           19.2              193.0   
279   280  Chinstrap  Dream            45.4           18.7              188.0   
280   281  Chinstrap  Dream            52.7           19.8              197.0   
..    ...        ...    ...             ...            ...                ...   
339   340  Chinstrap  Dream            55.8           19.8              207.0   
340   341  Chinstrap  Dream            43.5           18.1              202.0   
341   342  Chinstrap  Dream            49.6           18.2              193.0   
342   343  Chinstrap  Dream            50.8           19.0              210.0   
343   344  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  weight  
276       3500.0  female  2007    3.50  
277       3900.0    male  2007    3.90  
278       3650.0    male  2007    3.65  
279       3525.0  female  2007    3.52  
280       3725.0    male  2007    3.72  
..           ...     ...   ...     ...  
339       4000.0    male  2009    4.00  
340       3400.0  female  2009    3.40  
341       3775.0    male  2009    3.78  
342       4100.0    male  2009    4.10  
343       3775.0  female  2009    3.78  

[68 rows x 10 columns]

In [295]:
penguins[(penguins['species'] == "Chinstrap") & (penguins['island'] == "Dream")].count()

rowid                68
species              68
island               68
bill_length_mm       68
bill_depth_mm        68
flipper_length_mm    68
body_mass_g          68
sex                  68
year                 68
weight               68
dtype: int64

In [296]:
chinstrap_torgersen = penguins[(penguins['island'] == "Torgersen") & (penguins['species'] == "Chinstrap")] 

In [297]:
chinstrap_torgersen

Empty DataFrame
Columns: [rowid, species, island, bill_length_mm, bill_depth_mm, flipper_length_mm, body_mass_g, sex, year, weight]
Index: []

In [298]:
penguins[(penguins['species'] == "Chinstrap") & (penguins['island'] == "Torgersen")].count()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
weight               0
dtype: int64

<h1>INSIGHT NO.3</h1>

<H3> CHINSTRAP EXCLUSIVELY DREAM ISLAND</H3>


In [299]:
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0       1     Adelie  Torgersen        39.10000       18.70000   
1       2     Adelie  Torgersen        39.50000       17.40000   
2       3     Adelie  Torgersen        40.30000       18.00000   
3       4     Adelie  Torgersen        43.92193       17.15117   
4       5     Adelie  Torgersen        36.70000       19.30000   
..    ...        ...        ...             ...            ...   
339   340  Chinstrap      Dream        55.80000       19.80000   
340   341  Chinstrap      Dream        43.50000       18.10000   
341   342  Chinstrap      Dream        49.60000       18.20000   
342   343  Chinstrap      Dream        50.80000       19.00000   
343   344  Chinstrap      Dream        50.20000       18.70000   

     flipper_length_mm  body_mass_g     sex  year    weight  
0           181.000000  3750.000000    male  2007  3.750000  
1           186.000000  3800.000000  female  2007  3.800000  
2           195.000000  3250.000000  female  2007  3.250000  
3           200.915205  4201.754386    male  2007  4.201784  
4           193.000000  3450.000000  female  2007  3.450000  
..                 ...          ...     ...   ...       ...  
339         207.000000  4000.000000    male  2009  4.000000  
340         202.000000  3400.000000  female  2009  3.400000  
341         193.000000  3775.000000    male  2009  3.780000  
342         210.000000  4100.000000    male  2009  4.100000  
343         198.000000  3775.000000  female  2009  3.780000  

[344 rows x 10 columns]

<h1>INSIGHT NO.4</h1>

<H3> THE ISLAND DOES NOT IMPACT THE WEIGHT OF SPECIE ADELIE</H3>


<h1>ADELIE</h1>

In [300]:
adelie_Biscoe_w = np.round(adelie_Biscoe['weight'].mean(),1)

In [301]:
adelie_Biscoe_w 

np.float64(3.7)

In [302]:
adelie_dream_w = np.round(adelie_dream['weight'].mean(),1)

In [303]:
adelie_dream_w

np.float64(3.7)

In [304]:
adelie_torgersen_w = np.round(adelie_torgersen['weight'].mean(),1)

In [305]:
adelie_torgersen_w

np.float64(3.7)

<h1>INSIGHT NO.5</h1>

<H3> IMPACT WEIGHT</H3>


<h1>Gentoo</h1>

In [306]:
gentoo_biscoe_w = np.round(gentoo_biscoe['weight'].mean(),1)

In [307]:
gentoo_biscoe_w

np.float64(5.1)

In [308]:
gentoo_dream_w = np.round(gentoo_dream['weight'].mean(),1)

In [309]:
gentoo_dream_w

np.float64(nan)

In [310]:
gentoo_torgersen_w = np.round(gentoo_torgersen['weight'].mean(),1)

In [311]:
gentoo_torgersen_w

np.float64(nan)

<h1>INSIGHT NO.6</h1>

<H3> IMPACT WEIGHT</H3>


<h1>Chinstrap</h1>

In [312]:
chinstrap_biscoe_w = np.round(chinstrap_biscoe['weight'].mean(),1)

In [313]:
chinstrap_biscoe_w

np.float64(nan)

In [314]:
chinstrap_dream_w = np.round(chinstrap_dream['weight'].mean(),1)

In [315]:
chinstrap_dream_w

np.float64(3.7)

In [316]:
chinstrap_torgersen_w = np.round(chinstrap_torgersen['weight'].mean(),1)

In [317]:
chinstrap_torgersen_w

np.float64(nan)

<h1>INSIGHT NO.7</h1>

<H3> THE HEVIEST SPECIE IS GENTOO</H3>


<h1> HOW TO USE GROUPBY</h1>

<ol>
    <li> NAME THE GROUP BY REFERENCE DATA FRAME</li>
    <li>Call the dataframe</li>
    <li>.groupby function </li>
    <li>columns to group </li>
    <li>group by gunction </li>
</ol>

In [318]:
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0       1     Adelie  Torgersen        39.10000       18.70000   
1       2     Adelie  Torgersen        39.50000       17.40000   
2       3     Adelie  Torgersen        40.30000       18.00000   
3       4     Adelie  Torgersen        43.92193       17.15117   
4       5     Adelie  Torgersen        36.70000       19.30000   
..    ...        ...        ...             ...            ...   
339   340  Chinstrap      Dream        55.80000       19.80000   
340   341  Chinstrap      Dream        43.50000       18.10000   
341   342  Chinstrap      Dream        49.60000       18.20000   
342   343  Chinstrap      Dream        50.80000       19.00000   
343   344  Chinstrap      Dream        50.20000       18.70000   

     flipper_length_mm  body_mass_g     sex  year    weight  
0           181.000000  3750.000000    male  2007  3.750000  
1           186.000000  3800.000000  female  2007  3.800000  
2           195.000000  3250.000000  female  2007  3.250000  
3           200.915205  4201.754386    male  2007  4.201784  
4           193.000000  3450.000000  female  2007  3.450000  
..                 ...          ...     ...   ...       ...  
339         207.000000  4000.000000    male  2009  4.000000  
340         202.000000  3400.000000  female  2009  3.400000  
341         193.000000  3775.000000    male  2009  3.780000  
342         210.000000  4100.000000    male  2009  4.100000  
343         198.000000  3775.000000  female  2009  3.780000  

[344 rows x 10 columns]

In [319]:
penguins_agg = penguins.groupby(['species','sex','island','year']).size()

C:\Users\LAB4-StudentPC04\AppData\Local\Temp\ipykernel_7304\3861779661.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  penguins_agg = penguins.groupby(['species','sex','island','year']).size()


In [320]:
penguins_agg.unstack()

year                        2007  2008  2009
species   sex    island                     
Adelie    female Biscoe        5     9     8
                 Dream         9     8    10
                 Torgersen     8     8     8
          male   Biscoe        5     9     8
                 Dream        11     8    10
                 Torgersen    12     8     8
Chinstrap female Biscoe        0     0     0
                 Dream        13     9    12
                 Torgersen     0     0     0
          male   Biscoe        0     0     0
                 Dream        13     9    12
                 Torgersen     0     0     0
Gentoo    female Biscoe       16    22    20
                 Dream         0     0     0
                 Torgersen     0     0     0
          male   Biscoe       18    24    24
                 Dream         0     0     0
                 Torgersen     0     0     0

<h1>INSIGHT NO.8</h1>

<H3> THE HIGHEST RECORDED MALE ADELIE IS IN TORGERSEN</H3>


In [321]:
penguins_agg2 = np.round(penguins.groupby(['species','sex','island',])['weight'].mean(),2)

C:\Users\LAB4-StudentPC04\AppData\Local\Temp\ipykernel_7304\4263053565.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  penguins_agg2 = np.round(penguins.groupby(['species','sex','island',])['weight'].mean(),2)


In [322]:
penguins_agg3 = np.round(penguins.groupby(['species','sex'])['weight'].mean(),2)

C:\Users\LAB4-StudentPC04\AppData\Local\Temp\ipykernel_7304\3375569794.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  penguins_agg3 = np.round(penguins.groupby(['species','sex'])['weight'].mean(),2)


In [323]:
penguins_agg2.unstack()

island            Biscoe  Dream  Torgersen
species   sex                             
Adelie    female    3.37   3.34       3.40
          male      4.05   4.01       3.99
Chinstrap female     NaN   3.53        NaN
          male       NaN   3.94        NaN
Gentoo    female    4.68    NaN        NaN
          male      5.41    NaN        NaN

In [324]:
penguins_agg3.unstack()

sex        female  male
species                
Adelie       3.37  4.01
Chinstrap    3.53  3.94
Gentoo       4.68  5.41

<h1>INSIGHT NO.9</h1>

<H3> THE AGERAGE WEIGHT FOR ADELIE: </H3>
<H4>MALE: 4.01</H4>
<H4>FEMALE: 3.37</H4>



<h1>INSIGHT NO.10</h1>

<H3> THE AGERAGE WEIGHT FOR Chinstrap: </H3>
<H4>MALE: 3.94</H4>
<H4>FEMALE: 3.53</H4>



<h1>INSIGHT NO.11</h1>

<H3> THE AGERAGE WEIGHT FOR Gentoo: </H3>
<H4>MALE: 4.68</H4>
<H4>FEMALE: 5.41</H4>



<h1>DOES THE BILL LENGHT AFFECTS THE SIZE OF THE DEPTH</h1>

In [325]:
b_dept_lenght_cor = penguins['bill_length_mm'].corr(penguins['bill_depth_mm'])

In [326]:
b_dept_lenght_cor

np.float64(-0.23505287035553282)

<h1>INSIGHT NO. 12</h1>

<H3>BILL LENGHT HAS NO COORELATION WITH BILL DEPTH</H3>

<h1>DOES THE BILL LENGTH AFFECTS THE BODY MASS</h1>

In [327]:
blenght_body_cor = penguins['bill_length_mm'].corr(penguins['body_mass_g'])

In [328]:
blenght_body_cor

np.float64(0.5951098244376304)

<h1>INSIGHT NO. 13</h1>

<H3>BILL LENGHT HAS COORELATION WITH BODY MASS</H3>

<h1>DOES THE FLIPPER LENGTH AFFECTS THE BODY MASS</h1>

In [329]:
flenght_body_cor = penguins['flipper_length_mm'].corr(penguins['body_mass_g'])

In [330]:
flenght_body_cor

np.float64(0.8712017673060114)

<h1>INSIGHT NO. 14</h1>

<H3>FLIPPER LENGHT HAS COORELATION WITH BODY MASS</H3>

<h1>DOES THE FLIPPER AFFECTS THE BILL LENGTH</h1>

In [331]:
flenght_billlegnth_cor = penguins['flipper_length_mm'].corr(penguins['bill_length_mm'])

In [332]:
flenght_billlegnth_cor

np.float64(0.6561813407464278)

<h1>INSIGHT NO. 15</h1>

<H3>FLIPPER LENGHT HAS COORELATION WITH BILL LENGTH</H3>

<h1>CORRELATION BETWEEN QUALITATIVE AND QUANTITATIVE</h1>

<H3>SEX AND BODY MASS</H3>

In [333]:
pip install scipy


Note: you may need to restart the kernel to use updated packages.


In [334]:
from scipy.stats import f_oneway

<h3> requirements for one way annova </h3>
<h3> filter columns and its data </h3>

In [339]:
penguins[penguins['sex'] == 'male']['body_mass_g']

0      3750.000000
3      4201.754386
5      3650.000000
7      4675.000000
8      3475.000000
          ...     
334    3800.000000
336    3950.000000
339    4000.000000
341    3775.000000
342    4100.000000
Name: body_mass_g, Length: 179, dtype: float64

In [337]:
penguins[penguins['sex'] == 'female']['body_mass_g']

1      3800.0
2      3250.0
4      3450.0
6      3625.0
12     3200.0
        ...  
335    3525.0
337    3650.0
338    3650.0
340    3400.0
343    3775.0
Name: body_mass_g, Length: 165, dtype: float64

In [344]:
f_oneway(penguins[penguins['sex'] == 'male']['body_mass_g'],penguins[penguins['sex'] == 'female']['body_mass_g'])

F_onewayResult(statistic=np.float64(68.38476171974526), pvalue=np.float64(3.0226600487050246e-15))

In [341]:
f_stats, p_value =  f_oneway(penguins[penguins['sex'] == 'male']['body_mass_g'],penguins[penguins['sex'] == 'female']['body_mass_g'])

In [342]:
f_stats

np.float64(68.38476171974526)

In [343]:
p_value

np.float64(3.0226600487050246e-15)

<h1>INSIGHT NO. 16</h1>

<H3>THERE IS A CORRELATION BETWEEN SEX AND BODY MASS</H3>

In [345]:
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0       1     Adelie  Torgersen        39.10000       18.70000   
1       2     Adelie  Torgersen        39.50000       17.40000   
2       3     Adelie  Torgersen        40.30000       18.00000   
3       4     Adelie  Torgersen        43.92193       17.15117   
4       5     Adelie  Torgersen        36.70000       19.30000   
..    ...        ...        ...             ...            ...   
339   340  Chinstrap      Dream        55.80000       19.80000   
340   341  Chinstrap      Dream        43.50000       18.10000   
341   342  Chinstrap      Dream        49.60000       18.20000   
342   343  Chinstrap      Dream        50.80000       19.00000   
343   344  Chinstrap      Dream        50.20000       18.70000   

     flipper_length_mm  body_mass_g     sex  year    weight  
0           181.000000  3750.000000    male  2007  3.750000  
1           186.000000  3800.000000  female  2007  3.800000  
2           195.000000  3250.000000  female  2007  3.250000  
3           200.915205  4201.754386    male  2007  4.201784  
4           193.000000  3450.000000  female  2007  3.450000  
..                 ...          ...     ...   ...       ...  
339         207.000000  4000.000000    male  2009  4.000000  
340         202.000000  3400.000000  female  2009  3.400000  
341         193.000000  3775.000000    male  2009  3.780000  
342         210.000000  4100.000000    male  2009  4.100000  
343         198.000000  3775.000000  female  2009  3.780000  

[344 rows x 10 columns]

In [346]:
penguins[penguins['species'] == 'Adelie']['body_mass_g']

0      3750.000000
1      3800.000000
2      3250.000000
3      4201.754386
4      3450.000000
          ...     
147    3475.000000
148    3450.000000
149    3750.000000
150    3700.000000
151    4000.000000
Name: body_mass_g, Length: 152, dtype: float64

In [347]:
penguins[penguins['species'] == 'Gentoo']['body_mass_g']

152    4500.000000
153    5700.000000
154    4450.000000
155    5700.000000
156    5400.000000
          ...     
271    4201.754386
272    4850.000000
273    5750.000000
274    5200.000000
275    5400.000000
Name: body_mass_g, Length: 124, dtype: float64

In [348]:
penguins[penguins['species'] == 'Chinstrap']['body_mass_g']

276    3500.0
277    3900.0
278    3650.0
279    3525.0
280    3725.0
        ...  
339    4000.0
340    3400.0
341    3775.0
342    4100.0
343    3775.0
Name: body_mass_g, Length: 68, dtype: float64

In [351]:
f_stats_species, p_value_species =  f_oneway(penguins[penguins['species'] == 'Adelie']['body_mass_g'],
                                             penguins[penguins['species'] == 'Gentoo']['body_mass_g'],
                                            penguins[penguins['species'] == 'Chinstrap']['body_mass_g'])

In [352]:
f_stats_species

np.float64(338.5729100782222)

In [353]:
p_value_species

np.float64(1.0069770504490335e-81)